history.go

```diff
@@ -52,25 +57,33 @@ func Run() error {
 		if err != nil {
 			return fmt.Errorf("error reading the input")
 		}
-		//fmt.Println(text)
 		if text == "exit" || text == "quit" {
 			break
 		}
+		cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] command: %s", time.Now().Unix(), text))
```

Comment:
Let's leave the timestamp out for the time being; we just want to get the simplest possible thing fully working before adding refinements.

Reply:
I think it's fine. It is working already. Just applying small improvements.

Reply:
Right, but part of what you're learning is to do only enough engineering to solve the problem you actually need to solve. One of our biggest temptations as software engineers is to keep adding little bits here and there that seem like a nice idea, but this actually weakens the result, rather than strengthening it. Instead, we want to proceed by a logical, methodical workflow of:

1. Requiring a specific behaviour by means of a test
2. Implementing just enough code to pass the test
3. Refactoring the code for readability and simplicity

This takes a little practice to get completely used to, but it's worth making the effort to follow a really disciplined process.

Reply:
But I did that, I'm just still using grep. Or you're saying that I should refactor the tests before adding the timestamp?

Reply:
Nevermind! It is missing the tests for the Run function. So, you're right! Thank you!!!